src/linter/ui5Types/TypeLinter.ts

```diff
@@ -135,14 +146,43 @@ export default class TypeChecker {
 					this.#context, sourceFile.fileName,
 					sourceFile, sourceMaps,
 					checker, reportCoverage, messageDetails,
-					apiExtract, manifestContent
+					apiExtract, this.#filePathsWorkspace, manifestContent
+				);
+				await linter.lint();
+				linterDone();
+			}
+		}
+
+		// Will eventually produce new JS files for XML views and fragments
+		await lintXml({
+			filePathsWorkspace: this.#filePathsWorkspace,
+			workspace: this.#workspace,
+			context: this.#context,
+		});
```

Comment:
doesn't this lint all files again?

Reply:
Not all, but just the newly created ones. 
I have refactored it to respect the `xmlCompiledResource` flag: https://github.com/SAP/ui5-linter/pull/519/files#diff-1bc919fd57fdbc38b20c5feaca329ba1764224846e55d4a8b7b654f18e9ca35cR22
In essence, this means that the XML resource has already been linted, so we can skip it.

I was thinking in another direction- to provide a custom AbstractReader/Writer, but we need the "global" ones as inside we stash the transpiled XML->JS resources that are later being used in the SourceFileLinter